plugins/optimization-detective/optimization.php

```diff
@@ -172,6 +172,31 @@ function od_construct_preload_links( array $lcp_elements_by_minimum_viewport_wid
 	return implode( '', $preload_links );
 }
 
+/**
+ * Determines whether the response has an HTML Content-Type.
+ *
+ * @since n.e.x.t
+ * @private
+ *
+ * @return bool Whether Content-Type is HTML.
+ */
+function od_is_response_html_content_type(): bool {
+	$is_html_content_type = false;
+
+	$headers_list = array_merge(
+		array( 'Content-Type: ' . ini_get( 'default_mimetype' ) ),
+		headers_list()
+	);
+	foreach ( $headers_list as $header ) {
+		$header_parts = preg_split( '/\s*[:;]\s*/', strtolower( $header ) );
+		if ( count( $header_parts ) >= 2 && 'content-type' === $header_parts[0] ) {
+			$is_html_content_type = in_array( $header_parts[1], array( 'text/html', 'application/xhtml+xml' ), true );
```

Comment:
Note the `application/xhtml+xml` may be overkill per https://github.com/GoogleChromeLabs/wpp-research/pull/74, but I added it for completeness.